redis-sentinel/kontena.yml

```diff
@@ -0,0 +1,91 @@
+stack: kontena/redis-sentinel
+version: 1.0.0
+expose: lb
+description: Redis Sentinel Cluster with a Loadbalancer
+variables:
+  node_password:
+    type: string
+    from:
+      vault: ${STACK}-password
+      random_string: 16
+    to:
+      vault: ${STACK}-password
+  node_data_name:
+    type: string
+    default: "${STACK}-data"
+    from:
+      prompt: Redis data volume name
+  monitor_data_name:
+    type: string
+    default: "${STACK}-monitor"
+    from:
+      prompt: Redis monitor volume name
+services:
+  node:
+    image: redis:3.0-alpine
+    instances: 5
+    stateful: true
+    command: "/bin/sh -c 'redis-server --requirepass $$REDIS_SENTINEL_PASSWORD --masterauth $$REDIS_SENTINEL_PASSWORD'"
+    health_check:
+      protocol: tcp
+      port: 6379
+      initial_delay: 60
+      interval: 60
+    secrets:
+      - secret: ${STACK}-password
+        name: REDIS_SENTINEL_PASSWORD
+        type: env
+    environment:
+      KONTENA_LB_MODE: tcp
+      KONTENA_LB_EXTERNAL_PORT: 6379
+      KONTENA_LB_INTERNAL_PORT: 6379
+      KONTENA_LB_CUSTOM_SETTINGS: |
+        option tcp-check
+        tcp-check connect
+        tcp-check send AUTH\ {{ node_password }}\r\n
+        tcp-check expect string +OK
+        tcp-check send PING\r\n
+        tcp-check expect string +PONG
+        tcp-check send info\ replication\r\n
+        tcp-check expect string role:master
+        tcp-check send QUIT\r\n
+        tcp-check expect string +OK
+    links:
+      - lb
+    volumes:
+      - node-data:/data
+  monitor:
+    image: kontena/redis-sentinel:3.0-alpine
+    instances: 5
+    deploy:
+      wait_for_port: 26379
+    health_check:
+      protocol: tcp
+      port: 26379
+      initial_delay: 300
```

Comment:
BTW we definitely need to change health check reporting. 5 minutes to see service as unhealthy is somehow misleading when the reason is that health check is not run yet.